doozerlib/dblib.py

```diff
@@ -21,44 +148,128 @@ def __init__(self, runtime, environment, dry_run=False):
         :param dry_run: If True, not data will be written
         """
         self.lock = threading.Lock()
-        self.existing_domains = {}  # maps existing domain names to True
+
         self.runtime = runtime
         self.environment = environment
         self.cmd_line = ','.join(sys.argv)
         m = hashlib.md5()
         m.update(self.cmd_line.encode('utf-8'))
         self.uniq = m.hexdigest()  # help guarantee record uniqueness
 
-        self.client = None
-        if environment and not dry_run:
-            try:
-                self.client = boto3.client('sdb', region_name='us-east-1')
-                for name in self.client.list_domains()['DomainNames']:
-                    self.existing_domains[name] = True
-            except:
-                traceback.print_exc()
-                runtime.logger.error('Unable to acquire simpledb client; please check AWS creds')
-
-    def create_domain_if_missing(self, domain):
-        if self.client:
-            with self.lock:
-                if domain in self.existing_domains:
-                    # Already exists
-                    return
-                self.client.create_domain(
-                    DomainName=domain,
-                )
-                self.existing_domains[domain] = True
+        self.connection = None
+
+        self.column_identifier = self.create_column_identifier()
+
+    @staticmethod
+    def create_column_identifier():
+        int_handler = IntegerIdentifier()
+        float_handler = FloatIdentifier()
+        datetime_handler = DateTimeIdentifier()
+        varchar_handler = VarcharIdentifier()
+
+        int_handler.set_next_handler(float_handler).set_next_handler(datetime_handler).set_next_handler(varchar_handler)
+        return int_handler
+
+    def check_table_exist(self, table_name):
+        cursor = self.connection.cursor()
+        cursor.execute("""
+            SELECT COUNT(*)
+            FROM information_schema.tables
+            WHERE table_name = '{0}'
+            """.format(table_name.replace('\'', '\'\'')))
+        if cursor.fetchone()[0] == 1:
+            cursor.close()
+            return True
+
+        cursor.close()
+        return False
+
+    def get_missing_columns(self, attr_payload, table_name):
```

Comment:
On the first pass through this method, cache the list of columns. Use the cache for subsequent calls to the method  and only add if a new column name does not appear in that cache. Add the new column to the cache on the way out of the method.